### PR TITLE
Added the ability to mark controls as viewed

### LINKED ISSUE #1495: Add a way to mark controls as viewed
**Describe the feature you'd like**
Add the ability to mark a control as seen to help prevent loosing track of acknowledged controls.

**Describe alternatives you've considered**
A checkbox
Changing the background color of the control row (darker/lighter grey?)
Adding the option to hide the control, adding a toggle to show hidden controls

### PR DESCRIPTION
**Pull request by samuelsunvold**
Closes #1495. Add a way to mark controls as viewed.

Signed-off-by: Sam Sunvold <sunvoldsj@vcu.edu>

### CODE REVIEW

File: apps/frontend/src/components/cards/controltable/ControlRowHeader.vue [hunk 1]
```diff
@@ -132,6 +144,19 @@ export default class ControlRowHeader extends mixins(HtmlSanitizeMixin) {
...
+    return this.controlWasViewed;
+  }
+
+  set wasViewed(value: boolean) {
+    if(value) {
+      this.controlWasViewed = true;
+    } else {
+      this.controlWasViewed = false;
+    }
+  }
```
Comment (rbclark) [inline apps/frontend/src/components/cards/controltable/ControlRowHeader.vue:158]:
```suggestion
  set wasViewed(value: boolean) {
    this.controlWasWaived = value;
  }
```

File: apps/frontend/src/components/cards/controltable/ControlRowHeader.vue [hunk 2]
```diff
@@ -82,6 +82,14 @@
...
       </v-chip-group>
     </template>
+
+    <template #viewed>
+      <v-checkbox
+        v-model="wasViewed"
+        color="primary"
+        @click="$emit('control-viewed', control)"
+      />
+    </template>
```
Comment (rbclark) [inline apps/frontend/src/components/cards/controltable/ControlRowHeader.vue:92]:
```suggestion
    <template #viewed>
      <v-layout justify-center>
        <v-checkbox
          v-model="wasViewed"
          color="primary"
          hide-details
          class="align-center justify-center"
          @click="$emit('control-viewed', control)"
        />
      </v-layout>
    </template>
```

**Review by rbclark — CHANGES_REQUESTED** [review]
This may look better with the Viewed column centered:
Screen Shot 2021-06-24 at 6 31 04 AM

The headings are no longer properly aligned after this change:
Screen Shot 2021-06-24 at 6 12 57 AM


This breaks the layout on any screen size that isn't large:
Screen Shot 2021-06-24 at 6 34 08 AM
Screen Shot 2021-06-24 at 6 33 53 AM
Screen Shot 2021-06-24 at 6 33 41 AM

**Review by camdenmoors — APPROVED** [review]

**Comment by rbclark** [comment]
There's an invisible button on the right side of visibility that is throwing off the center alignment

Screen Shot 2021-06-24 at 11 23 39 AM

**Comment by aaronlippold** [comment]
Screen Shot 2021-06-29 at 11 05 57

This seems like a good model to borrow from ... if you need a reference ... click Files Changed on your PR :P

**Comment by aaronlippold** [comment]
I like the highlighted view checkbox and the running count of filtered elements

**Comment by rbclark** [comment]
> I like the highlighted view checkbox and the running count of filtered elements

You may want to clarify what you're looking for here, could you look at the netlify preview and describe what you want changed from there?

**Review by rbclark — CHANGES_REQUESTED** [review]
This PR seems to have a lot of unrelated changes showing up in it, can you reconcile it to only the required changes?

**Review by rbclark — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/2308869/127240492-c9516ac0-0c8b-404b-86ba-715d41c7ac88.mov

This seems to break all other sorting

**Review by rbclark — APPROVED** [review]

**Pull request merged**